modin/pandas/concat.py

```diff
@@ -65,89 +65,16 @@ def concat(objs,
-        def name_incrementer(i):
-            val = i[0]
-            i[0] += 1
-            return val
-
-        i = [0]
-        objs = [
-            series_to_df(
-                obj, obj.name if obj.name is not None else name_incrementer(i))
-            if isinstance(obj, pandas.Series) else obj for obj in objs
-        ]
-
-    # Using concat on the columns and index is fast because they're empty,
-    # and it forces the error checking. It also puts the columns in the
-    # correct order for us.
-    final_index = \
-        pandas.concat([pandas.DataFrame(index=idx) for idx in all_index],
-                      axis=axis, join=join, join_axes=join_axes,
-                      ignore_index=ignore_index, keys=keys, levels=levels,
-                      names=names, verify_integrity=verify_integrity,
-                      copy=False).index
-    final_columns = \
-        pandas.concat([pandas.DataFrame(columns=col)
-                       for col in all_columns],
-                      axis=axis, join=join, join_axes=join_axes,
-                      ignore_index=ignore_index, keys=keys, levels=levels,
-                      names=names, verify_integrity=verify_integrity,
-                      copy=False).columns
-
-    # Put all of the DataFrames into Ray format
-    # TODO just partition the DataFrames instead of building a new Ray DF.
-    objs = [
-        DataFrame(obj)
-        if isinstance(obj, (pandas.DataFrame, pandas.Series)) else obj
-        for obj in objs
-    ]
-
-    # Here we reuse all_columns/index so we don't have to materialize objects
-    # from remote memory built in the previous line. In the future, we won't be
-    # building new DataFrames, rather just partitioning the DataFrames.
-    if axis == 0:
-        new_blocks = np.array([
-            _reindex_helper._submit(
-                args=tuple([
-                    all_columns[i], final_columns, axis,
-                    len(objs[0]._block_partitions)
-                ] + part.tolist()),
-                num_return_vals=len(objs[0]._block_partitions))
-            for i in range(len(objs)) for part in objs[i]._block_partitions
-        ])
-    else:
-        # Transposing the columns is necessary because the remote task treats
-        # everything like rows and returns in row-major format. Luckily, this
-        # operation is cheap in numpy.
-        new_blocks = np.array([
-            _reindex_helper._submit(
-                args=tuple([
-                    all_index[i], final_index, axis,
-                    len(objs[0]._block_partitions.T)
-                ] + part.tolist()),
-                num_return_vals=len(objs[0]._block_partitions.T))
-            for i in range(len(objs)) for part in objs[i]._block_partitions.T
-        ]).T
-
-    return DataFrame(
-        block_partitions=new_blocks, columns=final_columns, index=final_index)
+    objs = [DataFrame(obj) if not isinstance(obj, DataFrame) else
```

Comment:
Does this present a `pandas.Series` in the correct format? The name of the `Series` should become the name of the column, I believe.

Reply:
As it currently is, it handles every case correctly except concatenating a `pandas.DataFrame` to `pandas.Series` on `axis=0`, in which case pandas actually ignores the name of the Series. I've updated the PR to mirror this.